source/server/filter_chain_manager_impl.cc

```diff
@@ -225,13 +241,59 @@ void FilterChainManagerImpl::addFilterChains(
       ++new_filter_chain_size;
     }
 
-    addFilterChainForDestinationPorts(
-        destination_ports_map_,
-        PROTOBUF_GET_WRAPPED_OR_DEFAULT(filter_chain_match, destination_port, 0), destination_ips,
-        server_names, filter_chain_match.transport_protocol(),
-        filter_chain_match.application_protocols(), direct_source_ips,
-        filter_chain_match.source_type(), source_ips, filter_chain_match.source_ports(),
-        filter_chain_impl);
+    // If using the matcher, require usage of "name" field and skip building the index.
+    if (matcher_) {
+      if (filter_chain->name().empty()) {
+        throw EnvoyException(fmt::format(
+            "error adding listener '{}': \"name\" field is required when using a listener matcher",
+            address_->asString()));
+      }
+      auto [_, inserted] =
+          filter_chains_by_name_.try_emplace(filter_chain->name(), filter_chain_impl);
+      if (!inserted) {
+        throw EnvoyException(
+            fmt::format("error adding listener '{}': \"name\" field is duplicated with value '{}'",
+                        address_->asString(), filter_chain->name()));
+      }
+    } else {
+      auto createAddressVector = [](const auto& prefix_ranges) -> std::vector<std::string> {
```

Comment:
nit: create_address_vector

Reply:
This is actually a lambda (this code is moved as-is).

Reply:
I believe this green block is not modified other than the indent.
However, github UI fails to render the diff as we expected...